src/wh_nvm.c

```diff
@@ -87,6 +87,36 @@ int wh_Nvm_GetAvailable(whNvmContext* context,
             out_reclaim_size, out_reclaim_objects);
 }
 
+int wh_Nvm_AddObjectPessimistic(whNvmContext* context, whNvmMetadata *meta,
+    const uint8_t* data, whNvmSize dataLen)
+{
+    uint16_t availableObjects;
+    uint16_t reclaimObjects;
+    int ret;
+    uint32_t availableSize;
+    uint32_t reclaimSize;
+    if (context == NULL)
+        return WH_ERROR_BADARGS;
+    /* check if we have room for the object */
+    ret = wh_Nvm_GetAvailable(context, &availableSize, &availableObjects,
+        &reclaimSize, &reclaimObjects);
+    if (ret == 0) {
+        if (availableSize >= WOLFHSM_NVM_METADATA_LEN + dataLen &&
```

Comment:
Metadata len is not included in the available data space
```suggestion
        if (availableSize >=  dataLen &&
```